editor/components/block-icon-with-colors/index.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import BlockIcon from '../block-icon';
+
+export default function BlockIconWithColors( { iconObject, className, ...props } ) {
```

Comment:
I wonder if this should be a separate component or just a prop in `BlockIcon`?

Reply:
Block icon receives just the icon source, and it can already be complex (a function, a component, an element or string). I feel adding another type (an object with a specific structure) will make things even more complex.

Reply:
I guess what I don't like is that the `BlockIcon` component doesn't support the same things as the `icon` attribute of a block. From someone external to Gutenberg, you shouldn't have to know the shape of the `icon` you're using and which component supports it. You just want to show the block icon so you just feed it to the `BlockIcon` component.

Reply:
Hi @youknowriad, thank you for the feedback. I updated the code and now we use only BlockIcon component, let me know if you prefer this version.